processor/tailsamplingprocessor/internal/sampling/util.go

```diff
@@ -93,3 +93,16 @@ func invertHasInstrumentationLibrarySpanWithCondition(ilss ptrace.ScopeSpansSlic
 	}
 	return true
 }
+
+func SetAttrOnScopeSpans(data *TraceData, attrName string, attrKey string) {
```

Comment:
It would be nice to have a test for this function here, and perhaps a benchmark. This way, we can compare it in the future with implementations that find the root span and apply it to that only.